src/Hadrian/Oracles/TextFile.hs

```diff
@@ -73,12 +72,37 @@ lookupDependencies depFile file = do
         Just [] -> error $ "No source file found for file " ++ quote file
         Just (source : files) -> return (source, files)
 
--- | Read and parse a @.cabal@ file, caching and tracking the result.
-readCabalData :: Context -> Action CabalData
-readCabalData = askOracle . CabalFile
+newtype TextFile = TextFile FilePath
+    deriving (Binary, Eq, Hashable, NFData, Show, Typeable)
+type instance RuleResult TextFile = String
 
--- | Read and parse a @.cabal@ file recording the obtained 'PackageData',
--- caching and tracking the result. Note that unlike 'readCabalData' this
--- function resolves all Cabal configuration flags and associated conditionals.
-readPackageData :: Context -> Action PackageData
-readPackageData = askOracle . PackageDataFile
+newtype KeyValue = KeyValue (FilePath, String)
+    deriving (Binary, Eq, Hashable, NFData, Show, Typeable)
+type instance RuleResult KeyValue = Maybe String
+
+newtype KeyValues = KeyValues (FilePath, String)
+    deriving (Binary, Eq, Hashable, NFData, Show, Typeable)
+type instance RuleResult KeyValues = Maybe [String]
+
+-- | This oracle reads and parses text files to answer various queries, caching
+-- and tracking the results.
+textFileOracle :: Rules ()
```

Comment:
Same as for the other caches, it'd be nice to get a sliiiightly more detailed overview. "Text files" just seems a little too vague. It'd probably be a good idea to mention which text files this actually gets used on. (.cabal files I suppose, but maybe others?)